commit f635ae7a07f56db636ab557a5f24bbcaf0b1bcd9
Author: Glenn Strauss <gstrauss@gluelogic.com>
Date:   Thu Nov 24 22:45:53 2016 -0500

    [mod_auth] compile fix for Mac OS X XCode (fixes #2772)
    
    memcpy() may be a macro and gave error for missing arguement when
    CONST_STR_LEN() macro is used (which expands to two arguments)
    
    (thx ryandesign)
    
    x-ref:
      "mod_authn_file.c:683:56: error: too few arguments provided to function-like macro invocation (memcpy)"
      https://redmine.lighttpd.net/issues/2772

diff --git src/mod_authn_file.c src/mod_authn_file.c
index 85470ab..a494957 100644
--- src/mod_authn_file.c
+++ src/mod_authn_file.c
@@ -680,7 +680,7 @@ static handler_t mod_authn_file_htpasswd_basic(server *srv, connection *con, voi
                     li_tohex(ntlmhex,sizeof(ntlmhex),ntlmhash,sizeof(ntlmhash));
 
                     /*(reuse sample buffer for salt  (FYI: expect slen == 8))*/
-                    memcpy(sample, CONST_STR_LEN("$1$"));
+                    memcpy(sample, "$1$", sizeof("$1$")-1);
                     memcpy(sample+sizeof("$1$")-1, b, slen);
                     sample[sizeof("$1$")-1+slen] = '\0';
                    #if defined(HAVE_CRYPT_R)
